### PR TITLE
refactor: use scale subresource to update replica counts

### PR DESCRIPTION
**Pull request by TheSpiritXIII**
This PR helps us get rid of our Deployment and StatefulSet RBAC policies one day (in favor of only allowing the scale sub-resource).

### CODE REVIEW

File: charts/operator/templates/role.yaml [hunk 1]
```diff
@@ -88,6 +88,11 @@ rules:
   apiGroups: ["apps"]
   resourceNames: ["rule-evaluator"]
   verbs: ["get", "patch", "update"]
+- resources:
+  - deployments/scale
```
Comment (bernot-dev) [inline charts/operator/templates/role.yaml:92]: I'm confused...why add these permissions now, which are a subset of the permissions that already exist? Why not just scale back the existing permissions when the rest of the necessary changes are ready?

Comment (TheSpiritXIII) [reply charts/operator/templates/role.yaml:92]: I tried running this change without this permission and it failed. You would think they are a subset but that's not how it's implemented in Kubernetes...

File: pkg/operator/rules_test.go [hunk 2]
```diff
@@ -473,3 +477,157 @@ func TestScaleRuleConsumers(t *testing.T) {
...
+		body = patchOptions.SubResourceBody
+	}
+
+	return c.client.Patch(ctx, body, patch, &patchOptions.PatchOptions)
+}
+
+func extractScale(obj client.Object) (autoscalingv1.Scale, error) {
+	switch obj := obj.(type) {
+	case *appsv1.Deployment:
+		var replicas int32 = 1
```
Comment (bernot-dev) [inline pkg/operator/rules_test.go:553]: go/go-style/best-practices#vardeclinitialization

File: pkg/operator/rules_test.go [hunk 3]
```diff
@@ -473,3 +477,157 @@ func TestScaleRuleConsumers(t *testing.T) {
...
+	body := obj
+	if patchOptions.SubResourceBody != nil {
+		body = patchOptions.SubResourceBody
+	}
+
+	return c.client.Patch(ctx, body, patch, &patchOptions.PatchOptions)
+}
+
+func extractScale(obj client.Object) (autoscalingv1.Scale, error) {
+	switch obj := obj.(type) {
```
Comment (bernot-dev) [inline pkg/operator/rules_test.go:551]: Thought: I'm sad that this type switch is necessary. But it seems to be.

**Review by bernot-dev — APPROVED** [review]
LGTM

**Pull request merged**